packages/amplify-graphql-api-construct/src/internal/construct-tree.ts

```diff
@@ -0,0 +1,26 @@
+import { Stack } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+
+/**
+ * Given a scope, search up for the parent stack. This should be the nearest stack object.
+ * @param scope the scope to search up against.
+ * @returns the stack, if one can be found, else throws an error.
+ */
+export const getStackForScope = (scope: Construct, getRootStack = false): Stack => {
+  const stacksInHierarchy = scope.node.scopes.filter((parentScope) => 'templateOptions' in parentScope);
```

Comment:
is `templateOptions` option known to exist on all stacks and only on a stack resource in the scope?

Reply:
Full search results https://github.com/search?q=repo%3Aaws%2Faws-cdk%20templateOptions&type=code